src/printf.h

```diff
@@ -9,4 +11,30 @@ struct Field;
 
 std::string verify_format_string(const std::string &fmt, std::vector<Field> args);
 
+class IPrintable
+{
+public:
+  virtual ~IPrintable() { };
+  virtual uint64_t value() = 0;
+};
+
+class PrintableString : public virtual IPrintable
+{
+public:
+  PrintableString(std::string value) : _value(value) { }
```

Comment:
This constructor could be changed to take `const std::string &value` to reduce the amount of copying we're doing here

Reply:
Or use std::move if you want to get fancy